dfttopif/parsers/pwscf.py

```diff
@@ -15,6 +16,25 @@ def __init__(self, directory):
         super(PwscfParser, self).__init__(directory)
         self.settings = {}
         parser = PwscfStdOutputParser()
+
+        # Look for appropriate files
+        try:
+            self.inputf = self.outputf = None
+            files = [f for f in os.listdir(self._directory) if os.path.isfile(os.path.join(directory, f))]
+            for f in files:
+                if self._get_line('Program PWSCF', f, basedir=self._directory, return_string=False):
+                    self.outputf = f
+                elif self._get_line('&control', f, basedir=self._directory, return_string=False, case_sens=False):
+                    self.inputf = f
+        except Exception as e:
+            raise InvalidIngesterException('Parser failed due to: ' + str(e))
```

Comment:
Is this necessarily an `InvalidIngesterException`?  I could imagine other filesystem-related exceptions being raised in that block that don't have anything to do with the ingester.

Reply:
It is not. I'll limit this to catching the Unicode error `_get_line` originally raised so that we don't catch anything else. Sound reasonable?

Reply:
You want to catch that on a file by file basis though, right?  Let's say a directory has some files that cause unicode errors, but also has a valid pwscf output file.  I think you want to just pass on that file in the files loop, not exit the initializer outright.

Reply:
That makes much more sense!